<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ANTSRegistration</class>
 <widget class="qMRMLWidget" name="ANTSRegistration">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>325</width>
    <height>451</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="ctkCollapsibleButton" name="inputsCollapsibleButton">
     <property name="text">
      <string>Inputs</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Fixed volume:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="fixedSelector">
        <property name="toolTip">
         <string>The fixed image</string>
        </property>
        <property name="nodeTypes">
         <stringlist notr="true">
          <string>vtkMRMLScalarVolumeNode</string>
         </stringlist>
        </property>
        <property name="showChildNodeTypes">
         <bool>false</bool>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="removeEnabled">
         <bool>false</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
        <property name="SlicerParameterName" stdset="0">
         <string notr="true">fixedVolume</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Moving volume:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qMRMLNodeComboBox" name="movingSelector">
        <property name="toolTip">
         <string>The moving image</string>
        </property>
        <property name="nodeTypes">
         <stringlist notr="true">
          <string>vtkMRMLScalarVolumeNode</string>
         </stringlist>
        </property>
        <property name="showChildNodeTypes">
         <bool>false</bool>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="removeEnabled">
         <bool>false</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
        <property name="SlicerParameterName" stdset="0">
         <string notr="true">movingVolume</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Initial transform:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="qMRMLNodeComboBox" name="initialSelector">
        <property name="toolTip">
         <string>Initial transform approximately places moving image onto the fixed image grid</string>
        </property>
        <property name="nodeTypes">
         <stringlist notr="true">
          <string>vtkMRMLLinearTransformNode</string>
          <string>vtkMRMLBSplineTransformNode</string>
          <string>vtkMRMLGridTransformNode</string>
         </stringlist>
        </property>
        <property name="showChildNodeTypes">
         <bool>true</bool>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="removeEnabled">
         <bool>true</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
        <property name="SlicerParameterName" stdset="0">
         <string notr="true">initialTransform</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="outputsCollapsibleButton">
     <property name="text">
      <string>Outputs</string>
     </property>
     <layout class="QFormLayout" name="formLayout_4">
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Transform:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="outputSelector">
        <property name="toolTip">
         <string>The resulting transform which can be used to resample moving image onto the fixed image grid</string>
        </property>
        <property name="nodeTypes">
         <stringlist notr="true">
          <string>vtkMRMLLinearTransformNode</string>
          <string>vtkMRMLBSplineTransformNode</string>
          <string>vtkMRMLGridTransformNode</string>
         </stringlist>
        </property>
        <property name="showChildNodeTypes">
         <bool>false</bool>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="addEnabled">
         <bool>true</bool>
        </property>
        <property name="removeEnabled">
         <bool>true</bool>
        </property>
        <property name="SlicerParameterName" stdset="0">
         <string notr="true">forwardTransform</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="advancedCollapsibleButton">
     <property name="text">
      <string>Parameters</string>
     </property>
     <property name="collapsed">
      <bool>false</bool>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Transform type:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="transformComboBox">
        <property name="toolTip">
         <string>The type of transform to use for registration process (ordered from simpler to more complicated)</string>
        </property>
        <property name="currentText">
         <string notr="true">Affine</string>
        </property>
        <property name="currentIndex">
         <number>3</number>
        </property>
        <item>
         <property name="text">
          <string>Translation</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Rigid</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Similarity</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Affine</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Affine metric: </string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="affineMetricWidget">
        <property name="toolTip">
         <string>The available metrics to use for registration process (ordered from simpler to more complicated)</string>
        </property>
        <property name="editable">
         <bool>false</bool>
        </property>
        <property name="currentText">
         <string notr="true">Mattes</string>
        </property>
        <property name="currentIndex">
         <number>4</number>
        </property>
        <item>
         <property name="text">
          <string>MeanSquares</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>CC</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>GC</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>JHMI</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Mattes</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Sampling rate:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="ctkSliderWidget" name="samplingRateSliderWidget">
        <property name="toolTip">
         <string>The fraction of voxels used to estimate the metric</string>
        </property>
        <property name="decimals">
         <number>3</number>
        </property>
        <property name="singleStep">
         <double>0.001000000000000</double>
        </property>
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="value">
         <double>0.200000000000000</double>
        </property>
        <property name="tickInterval">
         <double>0.050000000000000</double>
        </property>
        <property name="SlicerParameterName" stdset="0">
         <string>samplingRate</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>SyN metric:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QComboBox" name="synMetricWidget">
        <property name="toolTip">
         <string>The available metrics to use for registration process (ordered from simpler to more complicated)</string>
        </property>
        <property name="editable">
         <bool>false</bool>
        </property>
        <property name="currentText">
         <string notr="true">Mattes</string>
        </property>
        <property name="currentIndex">
         <number>4</number>
        </property>
        <item>
         <property name="text">
          <string>MeanSquares</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>CC</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>GC</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>JHMI</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Mattes</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Demons</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Gradient step:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="ctkDoubleSpinBox" name="gradientStepSpinBox">
        <property name="toolTip">
         <string>The size of the gradient step used by the optimizer</string>
        </property>
        <property name="decimals">
         <number>3</number>
        </property>
        <property name="minimum">
         <double>0.001000000000000</double>
        </property>
        <property name="maximum">
         <double>9.999000000000001</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.200000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="applyButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="toolTip">
      <string>Run the algorithm.</string>
     </property>
     <property name="text">
      <string>Apply</string>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkDoubleSpinBox</class>
   <extends>QWidget</extends>
   <header>ctkDoubleSpinBox.h</header>
  </customwidget>
  <customwidget>
   <class>ctkSliderWidget</class>
   <extends>QWidget</extends>
   <header>ctkSliderWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>ANTSRegistration</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>fixedSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>248</x>
     <y>61</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ANTSRegistration</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>outputSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>82</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>220</x>
     <y>161</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ANTSRegistration</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>movingSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>161</x>
     <y>8</y>
    </hint>
    <hint type="destinationlabel">
     <x>173</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
